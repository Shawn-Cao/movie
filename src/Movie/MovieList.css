.Movie {
  position: relative;
  width: inherit;
  padding: 20px 0;
  background-color: black;
}

.MovieList {
  display: inline-block;
  width: 154px;
  height: 278px;
}
