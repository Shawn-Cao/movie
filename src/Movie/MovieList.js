import React, { Component } from 'react';
import PropTypes from 'prop-types';
import MovieTile from './MovieTile';
import './MovieList.css';

class MovieList extends Component {
  static get propTypes() {
    return {
      movies: PropTypes.array
    };
  }

  render() {
    const movieTiles = this.props.movies.map((movie) =>
      <li key={movie.id} className='MovieList'><MovieTile movie={movie} /></li>
    );
    return (
      <ul className="Movie">{movieTiles}</ul>
    );
  }
}

export default MovieList;
