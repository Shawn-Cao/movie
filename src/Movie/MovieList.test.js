import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import MovieList from './MovieList';


const movies = [
  {
    "vote_count": 1724,
    "id": 297762,
    "video": false,
    "vote_average": 7,
    "title": "Wonder Woman",
    "popularity": 99.147603,
    "poster_path": "/gfJGlDaHuWimErCr5Ql0I8x9QSy.jpg",
    "original_language": "en",
    "original_title": "Wonder Woman",
    "genre_ids": [
      28,
      12,
      14,
      878
    ],
    "backdrop_path": "/hA5oCgvgCxj5MEWcLpjXXTwEANF.jpg",
    "adult": false,
    "overview": "An Amazon princess comes to the world of Man to become the greatest of the female superheroes.",
    "release_date": "2017-05-30"
  },
  {
    "vote_count": 4187,
    "id": 263115,
    "video": false,
    "vote_average": 7.5,
    "title": "Logan",
    "popularity": 68.113005,
    "poster_path": "/9EXnebqbb7dOhONLPV9Tg2oh2KD.jpg",
    "original_language": "en",
    "original_title": "Logan",
    "genre_ids": [
      28,
      18,
      878
    ],
    "backdrop_path": "/5pAGnkFYSsFJ99ZxDIYnhQbQFXs.jpg",
    "adult": false,
    "overview": "In the near future, a weary Logan cares for an ailing Professor X in a hideout on the Mexican border. But Logan's attempts to hide from the world and his legacy are upended when a young mutant arrives, pursued by dark forces.",
    "release_date": "2017-02-28"
  }
];

describe('MovieTime', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MovieList movies={movies}/>, div);
  });
  it('renders matching snapshot', () =>{
    const tree = renderer.create(
      <MovieList movies={movies}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
