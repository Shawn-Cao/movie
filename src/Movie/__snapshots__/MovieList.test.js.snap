// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieTime renders matching snapshot 1`] = `
<ul
  className="Movie"
>
  <li
    className="MovieList"
  >
    <div
      className="MovieTile"
    >
      <img
        alt="poster of movie: Wonder Woman"
        className="poster-img"
        src="http://image.tmdb.org/t/p/w185/gfJGlDaHuWimErCr5Ql0I8x9QSy.jpg"
      />
    </div>
  </li>
  <li
    className="MovieList"
  >
    <div
      className="MovieTile"
    >
      <img
        alt="poster of movie: Logan"
        className="poster-img"
        src="http://image.tmdb.org/t/p/w185/9EXnebqbb7dOhONLPV9Tg2oh2KD.jpg"
      />
    </div>
  </li>
</ul>
`;
